common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/Work.scala

```diff
@@ -175,6 +168,20 @@ object WorkState {
 
     override def id = canonicalId
   }
+
+  case class Derived(
+    sourceIdentifier: SourceIdentifier,
+    canonicalId: String,
+    modifiedTime: Instant,
+    derivedData: DerivedData,
+    relations: Relations[DataState.Identified] = Relations.none
+  ) extends WorkState {
+
+    type WorkDataState = DataState.Identified
+    type TransitionArgs = WorkData[WorkDataState]
```

Comment:
Hmm not really ideal that we have to pass this in as an arg. Maybe the `state` function in `Transition` should have access to the `WorkData` always instead?

Reply:
Yeah, I went through the same thought process - seemed like changing too much at the time but I'll give it a go now to see how it looks